internal/action/webhook.go

```diff
@@ -0,0 +1,55 @@
+// Copyright (c) 2021 - 2024, Ludvig Lundgren and the autobrr contributors.
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+package action
+
+import (
+	"bytes"
+	"context"
+	"net/http"
+	"time"
+
+	"github.com/autobrr/autobrr/internal/domain"
+	"github.com/autobrr/autobrr/pkg/errors"
+)
+
+func (s *service) webhook(ctx context.Context, action *domain.Action, release domain.Release) ([]string,error) {
+	s.log.Debug().Msgf("action WEBHOOK: '%s' file: %s", action.Name, release.TorrentName)
+
+	if len(action.WebhookData) > 1024 {
+		s.log.Trace().Msgf("webhook action '%s' - host: %s data: %s", action.Name, action.WebhookHost, action.WebhookData[:1024])
+	} else {
+		s.log.Trace().Msgf("webhook action '%s' - host: %s data: %s", action.Name, action.WebhookHost, action.WebhookData)
+	}
+
+	req, err := http.NewRequestWithContext(ctx, http.MethodPost, action.WebhookHost, bytes.NewBufferString(action.WebhookData))
+	if err != nil {
+		return nil,errors.Wrap(err, "could not build request for webhook")
+	}
+
+	req.Header.Set("Content-Type", "application/json")
+	req.Header.Set("User-Agent", "autobrr")
+
+	start := time.Now()
+	res, err := s.httpClient.Do(req)
+	if err != nil {
+		return nil,errors.Wrap(err, "could not make request for webhook")
+	}
+
+	defer res.Body.Close()
+
+	if len(action.WebhookData) > 256 {
+		s.log.Info().Msgf("successfully ran webhook action: '%s' to: %s payload: %s finished in %s with: %d", action.Name, action.WebhookHost, action.WebhookData[:256], time.Since(start), res.StatusCode)
+	} else {
+		s.log.Info().Msgf("successfully ran webhook action: '%s' to: %s payload: %s finished in %s with: %d", action.Name, action.WebhookHost, action.WebhookData, time.Since(start), res.StatusCode)
+	}
+
+	s.log.Info().Msgf("webhook ran and gave %i", res.StatusCode)
+	if res.StatusCode == 200 {
+		return nil, nil
+	} else if res.StatusCode == 201{
+		return []string{"Rejected release"}, nil
+	}
```

Comment:
I don't think we should universally treat HTTP 201 as a rejected status since we have no idea what users are doing for their existing webhooks, especially not because the action webhooks has had no concept of ok/fail except in the case it could not make the request and get a response.

The longer term plan has been to refactor the actions handling to be more like External Filters where you can set expected status etc and also add sub actions that can trigger depending on status. It's for sure a bigger task to do but would be a great improvement to users workflows.

Reply:
I'm happy to give this a go (in go), but would need some help on the typescript side of things, i.e. if the respective variables could be set in the UI and passed along to the webhook service, I can work on handling the issue.
What options would you envisage?
At the top of my head, I would expect the content and status code to be the most realistic scenarios, to distinguish between webhook outcomes. 
- Accept/Reject/Error depending on regex the content matches
- Accept/Reject/Error depending on status code
- Parse the resulting content as json and see if a given key-value combination matches. Determining Accept/Reject/Error based on key-value matches, as well as rejection reason.